x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/blocklist_form.tsx

```diff
@@ -354,6 +408,48 @@ export const BlockListForm = memo<ArtifactFormComponentProps>(
       [validateValues, onChange, item, blocklistEntry]
     );
 
+    const generateBlocklistEntryValue = useCallback(
+      (value: string | string[], newOperator: ListOperatorTypeEnum) => {
+        let result: string | string[] = [];
+
+        if (newOperator === ListOperatorTypeEnum.MATCH) {
+          if (Array.isArray(value) && value.length) {
+            result = value.join(',');
+          } else {
+            result = '';
+          }
+        } else {
+          if (typeof value === 'string' && value.length) {
+            result = value.split(',');
+          }
+        }
```

Comment:
this is called when the user doesn't really change the operator, but selects the same operator again. the result is that the value is cleared.

![bbbb](https://github.com/user-attachments/assets/951f7ca4-73f9-42cc-9bc3-d9240e0ba0e8)

i think it would be nice to address this, to keep the behaviour similar to the field dropdown

Reply:
Good catch, changed.

Reply:
empty string causes trouble here as well

![dddd](https://github.com/user-attachments/assets/6ea42b82-0e4d-4637-a109-065944ee26b9)